pkg/tfutils/tfWrapper.go

```diff
@@ -84,3 +86,52 @@ func runTfShowJson(directory string) (*State, error) {
 
 	return &state, nil
 }
+
+// function return true if resource identity is supported in the installed terraform version, false otherwise
+var isResourceIdentitySupported = func() (bool, error) {
```

Comment:
I think a variable with a inline function is not a good choice. This variable is global an could be changed by incident anywhere from the code. I think leaving the function as is and calling it is safer. I do not see why some sort of caching would be needed here.